backend/src/game.ts

```diff
@@ -0,0 +1,197 @@
+import { Express } from 'express';
+import * as code from './code';
+
+// for clarity, a gameID is just a string
+type UserId = string;
+// used ids for both players and host
+type GameId = string;
+
+// interface for user, mostly blank rn but will keep score or smth later.
+// Userid is stored in the map for now
+export interface User {
+	name: string;
+    answers: number[];
+}
+const usedGame: GameId[] = [];
+
+// define a quiz and question type
+export interface QuizQuestion {
+	questionText: string;
+	answerTexts: string[];
+	correctAnswers: number[];
+	note?: string;
+	time?: number;
+	points?: number;
+}
+
+export interface Quiz {
+	meta: {
+		title: string;
+		author: string;
+		pointDefault: number;
+		timeDefault: number;
+		note?: string;
+	};
+	questions: QuizQuestion[];
+}
+
+export interface Game {
+	users: Map<UserId, User>;
+	hostId: UserId;
+	activeQuestion: number;
+	quizData: Quiz;
+};
+
+function getUsers(game: Game) {
+	return [...(game.users.keys(), game.hostId)];
```

Comment:
This function is unused but I don't think it does what you expect. `(a, b)` in JS just evaluates to `b`, so this effectively just returns `[...game.hostId]` which breaks the host ID into an array of single characters. If you want to add it to the array you want something like:

```suggestion
	return [...game.users.keys(), game.hostId];
```